packages/styleguide/src/components/App/App.js

```diff
@@ -60,9 +62,31 @@ class App extends Component {
     this.handleNavLinkClick = this.handleNavLinkClick.bind(this);
   }
 
-  // eslint-disable-next-line class-methods-use-this
-  handleClick() {
-    this.setState({ isNavActive: !this.state.isNavActive });
+  showSidebar() {
+    // TODO: fix this weirdness (negated state inside `isActive` is because state is changed after handleClick function is called)
+    const isActive = !this.state.isNavActive;
+    const navigationBar = document.querySelector('.navigation-bar');
+    const currentScrollPosition = window.pageYOffset;
+
+    if (currentScrollPosition !== 0) {
+      navigationBar.style.transition = 'none';
```

Comment:
A ked to budes riesit cez context, celu touto logiku showSidebar mozes preniest do toho sidebaru.